pkg/agent/baker_test.go

```diff
@@ -18,80 +19,161 @@ func generateTestData() bool {
 	return os.Getenv("GENERATE_TEST_DATA") == "true"
 }
 
-var _ = Describe("Assert generated customData and cseCmd", func() {
-	DescribeTable("Generated customData and CSE", func(folder, k8sVersion string, configUpdator func(*datamodel.NodeBootstrappingConfiguration)) {
-		cs := &datamodel.ContainerService{
-			Location: "southcentralus",
-			Type:     "Microsoft.ContainerService/ManagedClusters",
-			Properties: &datamodel.Properties{
-				OrchestratorProfile: &datamodel.OrchestratorProfile{
-					OrchestratorType:    datamodel.Kubernetes,
-					OrchestratorVersion: k8sVersion,
+func getFakeContainerService(k8sVersion string) *datamodel.ContainerService {
+	cs := &datamodel.ContainerService{
+		Location: "southcentralus",
+		Type:     "Microsoft.ContainerService/ManagedClusters",
+		Properties: &datamodel.Properties{
+			TelemetryProfile: &datamodel.TelemetryProfile{
+				ApplicationInsightsKey: "c92d8284-b550-4b06-b7ba-e80fd7178faa",
+			},
+			OrchestratorProfile: &datamodel.OrchestratorProfile{
+				OrchestratorType:    datamodel.Kubernetes,
+				OrchestratorVersion: k8sVersion,
+				KubernetesConfig: &datamodel.KubernetesConfig{
+					ContainerRuntime:     "docker",
+					KubernetesImageBase:  "mcr.microsoft.com/oss/kubernetes/",
+					WindowsContainerdURL: "https://k8swin.blob.core.windows.net/k8s-windows/containerd/containerplat-aks-test-0.0.8.zip",
+					LoadBalancerSku:      "Standard",
+					CustomHyperkubeImage: "mcr.microsoft.com/oss/kubernetes/hyperkube:v1.16.15-hotfix.20200903",
+					ClusterSubnet:        "10.240.0.0/16",
+					NetworkPlugin:        "azure",
+					DockerBridgeSubnet:   "172.17.0.1/16",
+					ServiceCIDR:          "10.0.0.0/16",
+					EnableRbac:           to.BoolPtr(true),
+					EnableSecureKubelet:  to.BoolPtr(true),
+					KubeletConfig: map[string]string{
+						"--feature-gates": "RotateKubeletServerCertificate=true,a=b, PodPriority=true, x=y",
+					},
+					UseInstanceMetadata: to.BoolPtr(true),
+					DNSServiceIP:        "10.0.0.10",
+				},
+			},
+			HostedMasterProfile: &datamodel.HostedMasterProfile{
+				DNSPrefix: "uttestdom",
+				FQDN:      "uttestdom-dns-5d7c849e.hcp.southcentralus.azmk8s.io",
+				Subnet:    "10.240.0.0/16",
+			},
+			AgentPoolProfiles: []*datamodel.AgentPoolProfile{
+				{
```

Comment:
just to be clear we're not supposed to work with multiple agent pool. Please split them into different test cases.